### PR TITLE
chore: Initial setup for data collection 

### PR DESCRIPTION
**Pull request by ChaoticCross12**
#### Description of changes
This PR adds the first iteration of the E2E metrics script.

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/send-e2e-metrics.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import { CodeBuild } from 'aws-sdk';
+import { BuildBatch } from 'aws-sdk/clients/codebuild';
+import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
+import { PutCommand, DynamoDBDocumentClient } from '@aws-sdk/lib-dynamodb';
+import { getBatchesInProject, getBatchSourceVersionFromBatchId } from './collect-batch-id';
```
Comment (goldbez) [inline scripts/send-e2e-metrics.ts:5]: nit: rename this file to something more general like `codebuild-sdk-helpers`

File: package.json [hunk 2]
```diff
@@ -70,7 +70,8 @@
     "verify-api-extract": "yarn extract-api && ./scripts/verify-extract-api.sh",
     "verify-commit": "yarn ts-node ./scripts/verify-commit.ts",
     "verify-yarn-lock": "./scripts/verify-yarn-lock.sh",
-    "workflow-results": "yarn ts-node ./scripts/cci-workflow-results.ts"
+    "workflow-results": "yarn ts-node ./scripts/cci-workflow-results.ts",
+    "send-e2e-metrics": "yarn ts-node scripts/send-e2e-metrics.ts"
```
Comment (goldbez) [inline package.json:74]: This does not need to be in `package.json`

File: scripts/send-e2e-metrics.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import {
+  CostExplorerClient,
+  GetCostAndUsageCommand,
+  GetCostAndUsageCommandInput,
+  GetCostAndUsageCommandOutput,
+} from '@aws-sdk/client-cost-explorer';
+
+const getBatchBuildDetails = async (cb: CodeBuild, batchId: string[]): Promise<BuildBatch[]> => {
```
Comment (goldbez) [inline scripts/send-e2e-metrics.ts:13]: Should these helper functions go in `collect-batch-id.ts` (to be renamed)?

File: scripts/send-e2e-metrics.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      failFlag = false;
+      break;
+    }
+  }
+  if (failFlag) {
+    console.log('Could not find matching source version');
+    process.exit(1);
+  }
+
+  try {
```
Comment (goldbez) [inline scripts/send-e2e-metrics.ts:75]: This try-catch is too big. Better practice to wrap individual commands that might fall in their own try-catches.

**Review by goldbez — APPROVED** [review]

**Review by awsluja — APPROVED** [review]

**Pull request merged**